### PR TITLE
Repo cleanup + Upgrade testing CI env to Python v3.10

### PR DESCRIPTION
**Pull request by ayush9pandey**
There's a test failing for v3.6, which prompted me to add v3.10 for Github Actions CI. But adding another one would cause even longer delays on PRs as 3 more set of tests will be run. So I think we can remove v3.6 and instead have v3.10.

### CODE REVIEW

**Comment by ayush9pandey** [comment]
The last three tests will not run because 3.6 is removed now.

**Comment by ayush9pandey** [comment]
New CI workflow:

1. For all push commits on any branch except master, bioscrape will be installed and pytest will be run by the Github action called `test_package`.  This action runs only on ubuntu-latest and with Python v3.10 to keep the development workflow fast.
2. For all push commits directly to master (rare) and any pull requests to master, a detailed action called "deploy_package" will be run. This action installs bioscrape and runs pytest on all three OS and on three different Python versions to ensure backward compatibility. 

Since this PR has a new commit and it is a pull request to master, both actions are running and both have passed. So I feel that this PR is ready for review @WilliamIX . We need to merge this branch to master to make sure that this new actions workflow works for other PRs.

**Comment by WilliamIX** [comment]
3.6 and 3.7 don't seem to have run. Is this expected?

File: .github/workflows/deploy_bioscrape.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]  
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 4
```
Comment (WilliamIX) [inline .github/workflows/deploy_bioscrape.yml:11]: We can increase this number for faster runtime, right?

File: .github/workflows/deploy_bioscrape.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (WilliamIX) [inline .github/workflows/deploy_bioscrape.yml:14]: Can we keep 3.7?

File: .github/workflows/test_bioscrape.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+    branches: ["*","!master"]
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 4
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.10"]
```
Comment (WilliamIX) [inline .github/workflows/test_bioscrape.yml:12]: Can we do 3.10 and 3.7 here? Sandwich the versions?

**Review by WilliamIX — CHANGES_REQUESTED** [review]

File: .github/workflows/test_bioscrape.yml (hunk 3 above)
Comment (ayush9pandey) [reply .github/workflows/test_bioscrape.yml:12]: Yes. Done.

File: .github/workflows/deploy_bioscrape.yml (hunk 2 above)
Comment (ayush9pandey) [reply .github/workflows/deploy_bioscrape.yml:14]: Done.

File: .github/workflows/deploy_bioscrape.yml (hunk 1 above)
Comment (ayush9pandey) [reply .github/workflows/deploy_bioscrape.yml:11]: yes, I would guess so.

Comment (ayush9pandey) [reply .github/workflows/deploy_bioscrape.yml:11]:
```
By default, GitHub will maximize the number of jobs run in parallel depending on runner availability. 
```
We can remove the `max-parallel` argument then, it seems.

**Review by WilliamIX — APPROVED** [review]

**Comment by WilliamIX** [comment]
Because checks don't pass - we might need to merge this into dev before master?

**Comment by ayush9pandey** [comment]
do you see an option to merge the PR? It is blocked for me because 3.6 tests don't pass.

**Comment by ayush9pandey** [comment]
The problem is that the CI file needs to be in master to make sure the tests run. Even if we merge this to dev, and then when merging dev to master, we might see this problem again.

**Pull request merged**